src/simulators/matrix_product_state/matrix_product_state_internal.cpp

```diff
@@ -1345,35 +1431,28 @@ reg_t MPS::apply_measure(const reg_t &qubits,
   // since input is always sorted in qasm_controller, therefore, we must return the qubits 
   // to their original location (sorted)
   move_all_qubits_to_sorted_ordering();
-
-  reg_t outcome_vector_internal(qubits.size()), outcome_vector(qubits.size());
-  apply_measure_internal(qubits, rng, outcome_vector_internal);
-  for (uint_t i=0; i<qubits.size(); i++) {
-    outcome_vector[i] = outcome_vector_internal[i];
-  }
-  return outcome_vector;
+  return apply_measure_internal(qubits, rng);
 }
 
-void MPS::apply_measure_internal(const reg_t &qubits, 
-				  RngEngine &rng, reg_t &outcome_vector_internal) {
+reg_t MPS::apply_measure_internal(const reg_t &qubits, 
+				  RngEngine &rng) {
   reg_t qubits_to_update;
+  reg_t outcome_vector(qubits.size());
   for (uint_t i=0; i<qubits.size(); i++) {
-    outcome_vector_internal[i] = apply_measure(qubits[i], rng);
+    outcome_vector[i] = apply_measure_internal_single_qubit(qubits[i], rng);
   }
+  return outcome_vector;
 }
 
-uint_t MPS::apply_measure(uint_t qubit, 
-			 RngEngine &rng) {
+uint_t MPS::apply_measure_internal_single_qubit(uint_t qubit, RngEngine &rng) {	
   reg_t qubits_to_update;
   qubits_to_update.push_back(qubit);
 
   // step 1 - measure qubit 0 in Z basis
```

Comment:
I didn't understand the question. This method measures a single qubit whose index is `qubit`.

Reply:
I think the comment should say "measure the qubit in Z basis" instead of "measure qubit 0 in Z basis", because "qubit 0" can be another qubit, which we don't measure.

Reply:
ah, now I see the line you are referring to. Missed that. You are right, of course.